test/box-luatest/downgrade_test.lua

```diff
@@ -0,0 +1,799 @@
+    cg.server:exec(function()
+        local helper = require('test.box-luatest.downgrade_helper')
+
+        local test = box.schema.space.create('test', {engine = 'vinyl'})
+        local flags = box.space._space:get{test.id}.flags
+        flags.defer_deletes = true
+        box.space._space:update(test.id, {{'=', 'flags', flags}})
+
+        local app_version = helper.app_version('2.10.0')
+        t.assert_equals(box.schema.downgrade_issues(app_version), {})
+        box.schema.downgrade('2.10.0')
+        t.assert_equals(box.space._space:get{test.id}.flags.defer_deletes, true)
+
+        local prev_version = helper.prev_version('2.10.0')
+        t.assert_equals(box.schema.downgrade_issues(prev_version), {})
+        t.assert_equals(box.space._space:get{test.id}.flags.defer_deletes, true)
+
+        box.schema.downgrade(prev_version)
+        t.assert_equals(box.space._space:get{test.id}.flags.defer_deletes, nil)
+
+        -- idempotence check
+        box.schema.downgrade(prev_version)
+        t.assert_equals(box.space._space:get{test.id}.flags.defer_deletes, nil)
+    end)
+end
+
+-- Manual check:
+--
+--  Tarantool Enterprise 2.11.0-entrypoint-113-g803baaf
+--  type 'help' for interactive help
+--  tarantool> box.cfg{}
+--  tarantool> box.schema.space.create('city')
+--  tarantool> box.space.city:format{{name='payload', type='string'}}
+--  tarantool> box.space.city:create_index('payload')
+--  tarantool> box.space.city:replace{'moscow'}
+--  tarantool> box.schema.func.create('capitalize', {
+--      language = 'LUA',
+--      is_deterministic = true,
+--      body = [[function(row) error('boom') end]],
+--  })
+--  tarantool> box.space.city:upgrade{func = 'capitalize'}
+--  tarantool> box.schema.downgrade('2.9.1')
+--  ---
+--  - error: 'builtin/box/upgrade.lua:1861: Background update is active in space
+--      ''city''. It is supported starting from version 2.10.0.'
+--  ...
+g.test_downgrade_space_upgrade = function(cg)
+    t.tarantool.skip_if_not_enterprise()
+    cg.server:exec(function()
+        local helper = require('test.box-luatest.downgrade_helper')
+
+        local name = box.schema.space.create('name')
+        name:format{{name='payload', type='string'}}
+        name:create_index('payload')
+
+        local city = box.schema.space.create('city')
+        city:format{{name='payload', type='string'}}
+        city:create_index('payload')
+
+        name:replace{'ivan'}
+        city:replace{'moscow'}
+
+        box.schema.func.create('capitalize', {
+            language = 'LUA',
+            is_deterministic = true,
+            body = [[function(row) error('boom') end]],
+        })
+
+        local flags = {
+            upgrade = {
+                func = box.func.capitalize.id,
+                owner = box.info.uuid,
+            },
+        }
+        box.space._space:update(name.id, {{'=', 'flags', flags}})
+        box.space._space:update(city.id, {{'=', 'flags', flags}})
```

Comment:
You are not using Tarantool Enterprise here (i.e., you are not calling `:upgrade` method), why is this test skipped then? Same question for other Enterprise tests below.

Reply:
This `update` will trigger `upgrade` actually. We don't call `upgrade` directly to ensure the changes we do to the system spaces.